gostream/media.go

```diff
@@ -313,11 +313,14 @@ func (pc *producerConsumer[T, U]) start() {
 					}
 				}, err}
 				pc.currentMu.Unlock()
-				if lastRelease != nil {
-					lastRelease()
+				if prevRelease != nil {
+					prevRelease()
 				}
 			}()
 		}
+		if pc.current != nil && pc.current.Release != nil {
+			pc.current.Release()
+		}
```

Comment:
Add a comment above this explaining the need for releasing the current producer-consumer if it exists.